### PR TITLE
Edits to documentation per review feedback

### LINKED ISSUE #937: Update opening paragraph of dpctl docs
The opening paragraph of the documentation do not mention `dpctl.tensor` and primarily focus on the SYCL interface.

The paragraph should be modified to include detail on the `dpctl.tensor` module.

### LINKED ISSUE #938: User manual should have a section on dpctl.tensor
The "User manual" section table of contents (toc) of the docs right now do not include `dpctl.tensor`. A section about `dpctl.tensor` and `UsmMemory` classes should be added to the toc.

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Made explicit the intended use of event to ensure completion of submitted operation from host side of the application.

Removed note that `dpctl.lsplatform()` is new in dpctl 0.7

- [X] Have you provided a meaningful PR description?

Closes #937 
Closes #938 

@alexandraepan

### CODE REVIEW

File: docs/docfiles/user_guides/QuickStart.rst [hunk 1]
```diff
@@ -36,12 +36,9 @@ On Windows
...
 .. note::
 
-    The ``dpctl.lsplatform()`` function is new in dpctl 0.7 and will be
-    available in oneAPI 2021.3. If you are following the guide on an older
-    oneAPI installation, use ``dpctl.dump()``. If no GPU platforms are shown,
-    make sure your system has a supported GPU and has the necessary GPU drivers
-    installed. You can install GPU drivers by following the
-    `GPU driver installation guide`_.
+    If no GPU platforms are shown, make sure your system has a supported
+    GPU and has the necessary GPU drivers installed. You can install GPU
```
[the remark this answers is not in the document]
Comment (oleksandr-pavlyk) [reply docs/docfiles/user_guides/QuickStart.rst]: Thank you @alexandraepan This reads better indeed.

File: docs/docfiles/user_guides/QuickStart.rst [hunk 2]
```diff
@@ -36,12 +36,9 @@ On Windows
...
 
-    The ``dpctl.lsplatform()`` function is new in dpctl 0.7 and will be
-    available in oneAPI 2021.3. If you are following the guide on an older
-    oneAPI installation, use ``dpctl.dump()``. If no GPU platforms are shown,
-    make sure your system has a supported GPU and has the necessary GPU drivers
-    installed. You can install GPU drivers by following the
-    `GPU driver installation guide`_.
+    If no GPU platforms are shown, make sure your system has a supported
+    GPU and the necessary GPU drivers installed. See `GPU driver installation guide`_ to install GPU drivers.
+    drivers by following the `GPU driver installation guide`_.
```
Comment (diptorupd) [inline docs/docfiles/user_guides/QuickStart.rst:41]: This last line is incomplete and a repetition. Please remove.

**Review by diptorupd — COMMENTED** [review]
@oleksandr-pavlyk Since the doc is being touched, can you add a short few lines to address https://github.com/IntelPython/dpctl/issues/937. I will be good to have @alexandraepan review them in this PR itself.

File: docs/docfiles/user_guides/QuickStart.rst (hunk 2 above)
Comment (oleksandr-pavlyk) [reply docs/docfiles/user_guides/QuickStart.rst]: Done.

File: docs/docfiles/intro.rst [hunk 3]
```diff
@@ -3,16 +3,20 @@ Welcome to the Data-parallel Control (dpctl) Documentation!
...
 `Python buffer protocol <https://docs.python.org/3/c-api/buffer.html>`_
-using SYCL USM memory. It makes it possible to create Python objects that are
-backed by SYCL USM memory.
+
+Dpctl implements N-dimensional array Python object :class:`dpctl.tensor.usm_ndarray` to
+logically address USM memory allocations, and contains growing implementation of
+operations on such objects confofrming :array_api:`array API specification <>` in
+:class:`dpctl.tensor` namespace.
 
 Dpctl also supports the DPCPP :oneapi_filter_selection:`oneapi::filter_selector <>` extension and has
```
Comment (diptorupd) [inline docs/docfiles/intro.rst:20]: Let us change to DPC++ here too.

File: docs/docfiles/intro.rst [hunk 4]
```diff
@@ -3,16 +3,20 @@ Welcome to the Data-parallel Control (dpctl) Documentation!
...
 of the SYCL 2020 specification.
 
 Apart from the bindings for these runtime
 classes, dpctl includes bindings for SYCL USM memory allocators and
 deallocators. Dpctl Python API provides classes that implement
 `Python buffer protocol <https://docs.python.org/3/c-api/buffer.html>`_
-using SYCL USM memory. It makes it possible to create Python objects that are
-backed by SYCL USM memory.
+
+Dpctl implements N-dimensional array Python object :class:`dpctl.tensor.usm_ndarray` to
```
Comment (diptorupd) [inline docs/docfiles/intro.rst:15]: How about:

"Dpctl also includes a reference implementation for :array_api:`array API specification <>` using DPC++ and USM memory allocation under the :class:`dpctl.tensor` sub-module. The :class:`dpctl.tensor` sub-module provides an N-dimensional array Python object :class:`dpctl.tensor.usm_ndarray` and a growing implementation of :array_api:`array API specification <>` complaint operations on instances of the array class. "

**Review by diptorupd — CHANGES_REQUESTED** [review]

**Review by diptorupd — APPROVED** [review]

**Pull request merged**